### ISSUE TITLE
When car is connected after the planned shceduale it doesnt start charging.

### ISSUE THREAD

**Issue by t112013** [jonasbkarlsson/ev_smart_charging#172]
## Version of the custom_component

Version 1.8.0

## Configuration
Describe your configuration here.

## Describe the bug
If i connect the car after the planned-charging-time has passed the charging state remains off. My problem is that i controll the charger based on the charger switch state as you recomend in the documentation,  So every night i need to double check that the charger is Waiting for charging to begin or charging  and i take a look in the chargers values (i have easee charger), if not the charger stays off forever.  Turning the main switch off and on while car connected is on fixes the issue. 
Sometimes i even get "keeping the charger on" before the charging is supposed to start (i have "keep the charger on" since i use the hvac in the monring before leaving to work) this gives full power to the charger so it actually charges and i havnt dearded to make an automation to change the Amps to under 6 as recommended  by the easee integration since i dont reallt have an overview of all states of the sensor.
I think the integration should re-evalutate things when the "car connected" switches to on, of its maybe a bug from before ? 

## Debug log

```text

Add your logs here.

```

**Comment by jonasbkarlsson** [comment]
Do you have an automation thay sets the `switch.ev_smart_charging_ev_connected` in order to inform the integration that your car is connected to the charger? This is required to make the "keep the charger on" work, and can help in other cases as well.

**Comment by t112013** [comment]
Yes Yes ofcourse. So that car connected switch comes on and off based on the eaase charger beeing connected and so on. 
The other automation i run is the hooked to the charging switch as mentioned before. Charger on => sets easee charger to 10 A  charger to off => sets easee charger to 0 A

Both work great, as mentioned i think i just have issue when i connect the car after charging is supposed to start. i could try tonight again. Connect the car after the planned charging plan.

**Comment by t112013** [comment]
i Actually just managed to see the issue:
My time now is 14:44 
![Skjermbilde 2023-04-28 kl  14 43 19](https://user-images.githubusercontent.com/55782238/235151273-8099b04e-4a64-4d0f-9682-dfd4a13b9fba.png)

Charging is planned to 14:00 
![Skjermbilde 2023-04-28 kl  14 43 12](https://user-images.githubusercontent.com/55782238/235151417-a0678573-7177-407a-aec7-3923054305af.png)

change my switch of "car connected" to on.
The charging changed to on but status to waiting for Keeping charger on. 
![Skjermbilde 2023-04-28 kl  14 43 27](https://user-images.githubusercontent.com/55782238/235151571-628e807f-d694-4616-9634-efea938b246f.png)

Then if i toggle the switch from on to off and on again the charging goes again to on but status changes to charging.
![Skjermbilde 2023-04-28 kl  14 43 37](https://user-images.githubusercontent.com/55782238/235151641-6d1eed32-cd41-4318-9a08-2f608fefb351.png)

First night this happened to me i woke up to a working car with only 10 prosent of SOC
Normally we dont see this issue becasue maybe people have more SOC and charge at faster speed so the planned charging is much more later in the night like 2 AM isted of 20:00 of the same day ?
For instance car has 10% of SOC, i come home, forget to connect then charging is supposed to start at 20:00 then i go out and i plugg at 21:00, this behaviour happens to me, then car doesnt charge :P

**Comment by RienduPre** [comment]
It looks like I have the same issue with my Audi etron and Wallbox Pulsar plus

**Comment by t112013** [comment]
I would like to add that this happened to me again.
Came home wth about 70 prosent SOC, more or less arround 16:00, plugged the car,  integration got connected then integration had charging state on and keeping the charger on, so the car started charging, i manually turned it off, went to bed in the night and just hitted me to check that the car is chargning to my surprise it wasnt, integration had charrging planned false so no charging was going to happen.
The is something goign on when the integration calculates the charging plan and something happeneing to it when the car is connected too.
<img width="461" alt="Skjermbilde 2023-05-09 kl  00 01 54" src="https://user-images.githubusercontent.com/55782238/236948024-a8989e10-f532-4e03-8764-d8ec58703fef.png">
<img width="409" alt="Skjermbilde 2023-05-09 kl  00 01 15" src="https://user-images.githubusercontent.com/55782238/236948032-7d642296-f047-4f7b-9f63-1bd0ee431a39.png">

**Comment by t112013** [comment]
happened again. no charging, only way how to get it planning new charging was to turning the main swtich off and on 
<img width="1033" alt="Skjermbilde 2023-05-16 kl  22 50 32" src="https://github.com/jonasbkarlsson/ev_smart_charging/assets/55782238/9175f9f0-adc2-4c21-ae26-40a40cd6af40">
<img width="565" alt="Skjermbilde 2023-05-16 kl  22 50 44" src="https://github.com/jonasbkarlsson/ev_smart_charging/assets/55782238/0adf8cca-d303-4528-a4e0-93027bb98e5b">
<img width="928" alt="Skjermbilde 2023-05-16 kl  22 52 11" src="https://github.com/jonasbkarlsson/ev_smart_charging/assets/55782238/a93ba67b-7825-4f29-983f-9b68b9f6a65f">

**Comment by stejo388** [comment]
Hi,
I alos have seen this issue. Last time yesterday, is it only for me or do you also see this issue?

**Comment by jonasbkarlsson** [comment]
I noticed that @t112013 wrote "i manually turned it off". The manual on/off functions in the integration is completely independent of the automatic functions. So if the automatic functions has the charging on, and then you manually turn it off, the automatic function will not know about this. Could this be reason why it doesn't behave as you expect it to do?

**Comment by stejo388** [comment]
Yes I also turn off the automatic when my solar panels gives enough power through some automation. it should have been turned on again automatically but there could have been some problem. I will see if the problem comes back again and then dig into my automation. 

I think the reason for turning off the auto function and not just using manual start and stop was that otherwise the slot when the was not recalculated.
